src/Simplify.elm

```diff
@@ -853,6 +853,12 @@ Destructuring using case expressions
     Array.foldl (\_ soFar -> soFar) initial array
     --> initial
 
+    Array.toIndexedList Array.empty
+    --> []
+
+    List.map Tuple.second (Array.toIndexedList array)
+    --> Array.toList array
```

Comment:
Do you think we should also have this?
```elm
List.map Tuple.first (Array.toIndexedList array)
--> List.range 0 (Array.length array - 1)
```

Reply:
Would you want to add it? It should be more performant at least.

Reply:
I think it makes sense yeah. It's a bit simpler, shorter, more performant and it enables more optimizations with the `Array.length` checks we have.